File: 01-2.py
#!/usr/bin/env python

from functools import reduce

lines = [row.strip() for row in open('01.input').readlines()]

left = []
right = []

for line in lines:
    l, r = line.split()
    left.append(int(l))
    right.append(int(r))

left.sort()
right.sort()

print(reduce(lambda x, y: x + y, map(lambda l: l * right.count(l), left)))

File: 01-2.rb
#!/usr/bin/env ruby

lines = File.read('01.input').lines.map(&:strip)

left = []
right = []

lines.each do |line|
  l, r = line.split
  left << l.to_i
  right << r.to_i
end

puts left.map { |l| l * right.count(l) }.reduce &:+
